feat(legend): Add support for tile layers

milesap opened the pull request:
## Description
Adds a function `mapServerLegendToRendererAll` which is called when no `layerIndex` is provided to `mapServerToLocalLegend` for legend generation.

## Testing
testTileLegend.html

## Documentation
jsDoc

## Checklist
<!-- Quick checklist for items that are easy to miss -->

- [x] `gulp test` succeeds without warnings or errors
- [ ] release notes have been updated
- [x] all commit messages are descriptive and follow guidelines
- [x] PR targets the correct release version
- I will assign this PR to the primary reviewer

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/fgpv-vpgf/geoapi/185)
<!-- Reviewable:end -->

james-rae commented:
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/symbology.js, line 801 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/geoapi/185#-KfC7_zquGso40I4jAfM:-KfC7_zrz6yor-v10G1P:bvie7tm) ([raw file](https://github.com/fgpv-vpgf/geoapi/blob/a6ffecc740ad3b6f3ace5d300976b2163730deb8/src/symbology.js#L801)):*
> ```JavaScript
>      * @function mapServerToLocalLegend
>      * @param {String}    mapServerUrl  service url (root service, not indexed endpoint)
>      * @param {Integer}   layerIndex    the index of the layer in the legend we are interested in
> ```

Document that this is now optional, and that if it is not provided, all layers will be collapsed into a single legend

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/geoapi/185)*
<!-- Sent from Reviewable.io -->

milesap commented:
Review status: 1 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/symbology.js, line 801 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/geoapi/185#-KfC7_zquGso40I4jAfM:-KfC1PuDw89p_9NpWVms:bcxcw9h) ([raw file](https://github.com/fgpv-vpgf/geoapi/blob/a6ffecc740ad3b6f3ace5d300976b2163730deb8/src/symbology.js#L801)):*
<details><summary><i>Previously, james-rae (James Rae) wrote…</i></summary><blockquote>

Document that this is now optional, and that if it is not provided, all layers will be collapsed into a single legend
</blockquote></details>

Ok changed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/geoapi/185)*
<!-- Sent from Reviewable.io -->

james-rae commented:
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/geoapi/185)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
Reviewed 2 of 2 files at r1.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/geoapi/185)*
<!-- Sent from Reviewable.io -->

alyec commented:
Reviewed 1 of 2 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[src/symbology.js, line 801 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/geoapi/185#-KfC7_zquGso40I4jAfM:-KfD0XnqHJOcZrEROKan:b-2y1yho) ([raw file](https://github.com/fgpv-vpgf/geoapi/blob/a6ffecc740ad3b6f3ace5d300976b2163730deb8/src/symbology.js#L801)):*
<details><summary><i>Previously, james-rae (James Rae) wrote…</i></summary><blockquote>

Document that this is now optional, and that if it is not provided, all layers will be collapsed into a single legend
</blockquote></details>

`[layerIndex]`

---

*[src/symbology.js, line 774 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/geoapi/185#-KfD-4UkdFXt_8zPrLN9:-KfD-4UkdFXt_8zPrLNA:by662um) ([raw file](https://github.com/fgpv-vpgf/geoapi/blob/4768c15a9e7c9b983578480d87296edc1d923dcc/src/symbology.js#L774)):*
> ```JavaScript
> 
>     let uniqueValueInfos = [];
>     serverLegend.layers.forEach(layer => {
> ```

Slight variation with two map calls (mostly for style / readability):
```js
    const layerRenders = serverLegend.layers.map(layer =>
        layer.legend.map(layerLegend => ({
            label: layerLegend.label,
            symbol: {
                type: 'esriPMS',
                imageData: layerLegend.imageData,
                contentType: layerLegend.contentType
            }
        }))
    );
    return {
        type: 'uniqueValue',
        uniqueValueInfos: [].concat(...layerRenders);
    };
```

Personally I find things like the `.concat` easy to miss when there's a big block of code in between them, feel free to disagree and keep the original.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/geoapi/185)*
<!-- Sent from Reviewable.io -->

milesap commented:
Review status: 1 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/symbology.js, line 801 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/geoapi/185#-KfC7_zquGso40I4jAfM:-KfCEejqZk-f6znzkR3T:b499q7j) ([raw file](https://github.com/fgpv-vpgf/geoapi/blob/a6ffecc740ad3b6f3ace5d300976b2163730deb8/src/symbology.js#L801)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

`[layerIndex]`
</blockquote></details>

Changed

---

*[src/symbology.js, line 774 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/geoapi/185#-KfD-4UkdFXt_8zPrLN9:-KfCEawI1EV0p6hD3YsL:bo3pjd3) ([raw file](https://github.com/fgpv-vpgf/geoapi/blob/4768c15a9e7c9b983578480d87296edc1d923dcc/src/symbology.js#L774)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

Slight variation with two map calls (mostly for style / readability):
```js
    const layerRenders = serverLegend.layers.map(layer =>
        layer.legend.map(layerLegend => ({
            label: layerLegend.label,
            symbol: {
                type: 'esriPMS',
                imageData: layerLegend.imageData,
                contentType: layerLegend.contentType
            }
        }))
    );
    return {
        type: 'uniqueValue',
        uniqueValueInfos: [].concat(...layerRenders);
    };
```

Personally I find things like the `.concat` easy to miss when there's a big block of code in between them, feel free to disagree and keep the original.
</blockquote></details>

This looks better, changed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/geoapi/185)*
<!-- Sent from Reviewable.io -->

alyec commented:
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/geoapi/185)*
<!-- Sent from Reviewable.io -->

james-rae commented:
Reviewed 1 of 1 files at r3.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/geoapi/185)*
<!-- Sent from Reviewable.io -->
